### PR TITLE
MailChimp newsletter settings UI component.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information

This is a part of #17434 PR that represents MailChimp Store Informations UI view. This is the fourth step of the setup procedure.
This is provided for review as a separate PR because the original (spike) PR is too big for the review to be comfortable. 

To test this component in action please refer to #17434.

image

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import QueryMailChimpLists from 'woocommerce/state/sites/settings/email/queryLists';
+
+const NewsletterSettings = ( { storeData, onChange, siteId, isRequesting, translate } ) => {
+	return (
+		<FormFieldset className="setup-steps__store-info-field">
+			<QueryMailChimpLists siteId={ siteId } />
+			<div>Pick a list, you will not able to change it for now so pick carfully</div>
```
Comment (psealock) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: ~carfully~ `->` carefully

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			<div>Create you list at mailchimp.com if you have not done it aready</div>
```
Comment (psealock) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: ~you~ your

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js (hunk 1 above)
Comment (justinshreve) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: This string also needs wrapped in `translate`.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js (hunk 2 above)
Comment (justinshreve) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: This needs wrapped in `translate` as well.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { map } from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isRequestingLists } from 'woocommerce/state/sites/settings/email/selectors';
+import { localize } from 'i18n-calypso';
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: `i18n-calypso` is an external dependency.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isRequestingLists } from 'woocommerce/state/sites/settings/email/selectors';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import QueryMailChimpLists from 'woocommerce/state/sites/settings/email/queryLists';
+
+const NewsletterSettings = ( { storeData, onChange, siteId, isRequesting, translate } ) => {
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Same suggestion on `storeData` here, add a default `{}` and/or make isRequired in the prop types.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			</FormSelect>
+		</FormFieldset>
+	);
+};
+
+NewsletterSettings.propTypes = {
+	onChange: PropTypes.func.isRequired,
+	storeData: PropTypes.object,
+	siteId: PropTypes.number.isRequired,
+	isRequesting: PropTypes.bool
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Trailing comma

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+const NewsletterSettingsConnected = connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		return {
+			siteId,
+			isRequesting: isRequestingLists( state, siteId )
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Trailing comma

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import QueryMailChimpLists from 'woocommerce/state/sites/settings/email/queryLists';
+
+const NewsletterSettings = ( { storeData, onChange, siteId, isRequesting, translate } ) => {
+	return (
+		<FormFieldset className="setup-steps__store-info-field">
+			<QueryMailChimpLists siteId={ siteId } />
+			<div>{ translate( 'Pick a list, you will not able to change it for now so pick carefully' ) }</div>
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: Do we need periods at the end of this and the next sentence?

Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/newsletter-settings.js]: We do - they are full sentences they go with a period :+1:

**Pull request merged**